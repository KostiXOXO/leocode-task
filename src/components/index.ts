import UsersList from './list/UsersList';
import SearchBox from './searchBox/SearchBox';
import FiltrableUsersList from './filtrableUsersList/FiltrableUsersList';
import Spinner from './spinner/Spinner';

export { UsersList, SearchBox, FiltrableUsersList, Spinner };
