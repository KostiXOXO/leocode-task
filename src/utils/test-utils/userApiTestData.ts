export const sampleData = [
	{
		id: 1,
		name: 'test user',
		username: 'test_user1',
	},
	{
		id: 2,
		name: 'tetrisuser',
		username: 'noneman',
	},
];
