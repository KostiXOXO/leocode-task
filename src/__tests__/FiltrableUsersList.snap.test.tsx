import React from 'react';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { FiltrableUsersList } from 'components';
import { sampleData } from 'utils/test-utils/userApiTestData';

afterEach(cleanup);

it('Component did not change', async () => {
	const tree = renderer.create(<FiltrableUsersList data={sampleData} />).toJSON();
	expect(tree).toMatchSnapshot();
});
