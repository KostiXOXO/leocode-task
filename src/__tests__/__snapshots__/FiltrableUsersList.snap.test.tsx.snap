// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component did not change 1`] = `
<div
  className="filtrableUsersListContainer"
>
  <form
    className="userInputForm"
  >
    <input
      aria-label="searchbox-input-field"
      data-testid="input-field"
      onChange={[Function]}
      placeholder="Search by user name..."
      type="text"
      value=""
    />
  </form>
  <ol
    className="listContainer ordered"
    data-testid="listComponent"
  />
</div>
`;
